src/network/networkd-netlabel.c

```diff
@@ -0,0 +1,213 @@
+                r = sd_netlink_message_append_in6_addr(m, NLBL_UNLABEL_A_IPV6ADDR, &address->in_addr.in6);
+                if (r < 0)
+                        return r;
+
+                r = sd_netlink_message_append_in6_addr(m, NLBL_UNLABEL_A_IPV6MASK, &netmask.in6);
+        }
+        if (r < 0)
+                return r;
+
+        return sd_netlink_call_async(address->link->manager->genl, NULL, m, netlabel_handler, NULL, address->link, 0, NULL);
+}
+
+static void address_add_netlabel_set(const Address *address, Set *labels) {
+        _cleanup_free_ char *addr_str = NULL;
+        int r;
+        const char *label;
+
+        (void) in_addr_prefix_to_string(address->family, &address->in_addr, address->prefixlen, &addr_str);
+
+        SET_FOREACH(label, labels) {
+                r = netlabel_command(NLBL_UNLABEL_C_STATICADD, label, address);
+                if (r < 0)
+                        log_link_warning_errno(address->link, r, "Adding NetLabel %s for IP address %s failed, ignoring: %m",
+                                               label, strna(addr_str));
+                log_link_debug(address->link, "Added NetLabel %s for IP address %s",
+                               label, strna(addr_str));
+        }
+}
+
+void address_add_netlabel(const Address *address) {
+        assert(address);
+        assert(address->link);
+
+        if (!address->link->network || !IN_SET(address->family, AF_INET, AF_INET6))
+                return;
+
+        if (address->family == AF_INET && IN_SET(address->source, NETWORK_CONFIG_SOURCE_DHCP4, NETWORK_CONFIG_SOURCE_DHCP_PD) && address->link->network->dhcp4_netlabels)
+                address_add_netlabel_set(address, address->link->network->dhcp4_netlabels);
+        else if (address->family == AF_INET6 && IN_SET(address->source, NETWORK_CONFIG_SOURCE_DHCP6, NETWORK_CONFIG_SOURCE_DHCP_PD) && address->link->network->dhcp6_netlabels)
+                address_add_netlabel_set(address, address->link->network->dhcp6_netlabels);
+        else if (IN_SET(address->source, NETWORK_CONFIG_SOURCE_STATIC))
+                address_add_netlabel_set(address, address->netlabels);
+}
+
+void address_del_netlabel(const Address *address) {
+        int r;
+        _cleanup_free_ char *addr_str = NULL;
+
+        assert(address);
+        assert(address->link);
+
+        if (!address->link->network || !IN_SET(address->family, AF_INET, AF_INET6))
+                return;
+
+        (void) in_addr_prefix_to_string(address->family, &address->in_addr, address->prefixlen, &addr_str);
+
+        r = netlabel_command(NLBL_UNLABEL_C_STATICREMOVE, NULL, address);
+        if (r < 0)
+                log_link_warning_errno(address->link, r, "Deleting NetLabels for IP address %s failed, ignoring: %m",
+                                       strna(addr_str));
+        log_link_debug(address->link, "Deleted NetLabels for IP address %s",
+                       strna(addr_str));
```

Comment:
Same here, the debugging log is added even `netlabel_command()` fails.

Also, `Deleted` -> `Deleting` or `Removing`